### ISSUE TITLE
Client API: full_return paramenter missing from cmd_subset function

### ISSUE THREAD

**Issue by mvintila** [saltstack/salt#47000]
### Description of Issue/Question
If cmd_subset function is called with the 'full_return' paramenter, it will not match any minions. If called without that parameter it works as expected.

This is caused by the said parameter missing from the function declaration in salt/client/__init__.py after line 441 (https://github.com/saltstack/salt/blob/develop/salt/client/__init__.py#L441). Because it is missing, it is passed as generic **kwargs to self.cmd function call on line 466, which modifies the output and prevents the whole thing from working.

This is very simple to fix by adding the said parameter in the declaration and in the return function after line 481.

### Setup

```
import salt.client
salt = salt.client.LocalClient(c_path='/etc/salt/master')
results = salt.cmd_subset('*', 'cmd.run', ['hostname'], sub=1, full_return=True)
print(results)
```

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)
Run the above code and notice that no minions will be matched. Run the same code without the 'full_return=True' , and notice it will work as expected.

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
Latest salt installed with pyp, python 2.7 in vritualenv:

Salt Version:
           Salt: 2018.3.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Dec  4 2017, 14:50:18)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 17.0.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.1.6
 
System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-78-generic
         system: Linux
        version: Ubuntu 16.04 xenial

**Comment by gtmanfred** [comment]
Since you have already found how to fix this, can you submit this change as a bug fix PR against 2017.7, and we will merge it forward?

Thanks,
Daniel

**Comment by rallytime** [comment]
Closed via #47055
